src/main/scala/edu/ie3/mobsim/MobilitySimulator.scala

```diff
@@ -411,38 +419,45 @@ final class MobilitySimulator(
       availableChargingPoints: Map[UUID, Int],
       maxDistance: ComparableQuantity[Length]
   ): Future[Option[(UUID, Movement)]] = Future {
-    chooseChargingStation(
+    val (ratingMap, evOption) = chooseChargingStation(
       ev,
       pricesAtChargingStation,
       availableChargingPoints,
       MobilitySimulator.seed,
       maxDistance
-    ).map { cs =>
-      val availableChargingPointsAtStation: Int =
-        availableChargingPoints.getOrElse(cs, 0)
-      if (availableChargingPointsAtStation > 0) {
-        ev.setChargingAtSimona(true)
-        ev.setChosenChargingStation(Some(cs))
+    )
 
-        logger.debug(
-          s"${ev.getId} starts charging at $cs."
-        )
+    ratingMap
+      .map { cs =>
+        val availableChargingPointsAtStation: Int =
+          availableChargingPoints.getOrElse(cs, 0)
+        if (availableChargingPointsAtStation > 0) {
+
+          val updatedEv = evOption
+            .getOrElse(ev)
+            .setChargingAtSimona()
+            .setChosenChargingStation(Some(cs))
 
-        Some((cs, Movement(cs, ev)))
-      } else {
+          logger.debug(
+            s"${ev.getId} starts charging at $cs."
+          )
+
+          Some((cs, Movement(cs, updatedEv)))
```

Comment:
Since the `cs` UUID is already saved within movement, this method can easily return `Future[Option[Movement]]` only